lib/cocoapods-core/cdn_source.rb

```diff
@@ -369,133 +404,118 @@ def download_and_save_with_retries_async(partial_url, file_remote_url, etag, ret
+      case response.status
+      when 301
+        redirect_location = response.headers['location']
+        debug "CDN: #{name} Redirecting from #{file_remote_url} to #{redirect_location}"
+        download_and_save_with_retries_async(partial_url, redirect_location, etag)
+      when 304
+        debug "CDN: #{name} Relative path not modified: #{partial_url}"
+        # We need to update the file modification date, as it is later used for freshness
+        # optimization. See #initialize for more information.
+        FileUtils.touch path
+      when 200
+        File.open(path, 'w') do |f|
+          encoding = response.headers['content-encoding'].to_s
+          if encoding.present?
+            case encoding
+            when 'gzip'
+              body = Zlib::GzipReader.wrap(StringIO.new(body), &:read)
+            else
+              raise Informative, "CDN: #{name} URL couldn't be saved: #{file_remote_url} Content encoding: #{response.headers['content-encoding']}"
             end
           end
+
+          f.write(body&.force_encoding('UTF-8'))
+        end
+
+        etag_new = response.headers['etag']
+        debug "CDN: #{name} Relative path downloaded: #{partial_url}, save ETag: #{etag_new}"
+        File.open(etag_path, 'w') { |f| f.write(etag_new) } unless etag_new.nil?
+      when 404
+        debug "CDN: #{name} Relative path couldn't be downloaded: #{partial_url} Response: #{response.status}"
+        nil
+      when 502, 503, 504
+        # Retryable HTTP errors, usually related to server overloading
+        message = "CDN: #{name} URL couldn't be downloaded: #{file_remote_url} Response: #{response.status} #{body}"
+        if retries <= 1
+          raise Informative, message
         else
-          raise Informative, "CDN: #{name} URL couldn't be downloaded: #{file_remote_url} Response: #{response.response_code} #{response.response_body}"
+          debug message + ", retries: #{retries - 1}"
+          make_sleep backoff_time(retries)
+          download_and_save_with_retries_async(partial_url, file_remote_url, etag, retries - 1)
         end
+      else
+        raise Informative, "CDN: #{name} URL couldn't be downloaded: #{file_remote_url} Response: #{response.status} #{body}"
       end
+    end
+
+    def make_download(file_remote_url, etag = nil)
+      headers = [
+        %w[Accept-Encoding gzip],
+      ]
 
-      # Calling `Future#run` flattens the chained futures created by retries or redirects
-      #
-      # Does not, in fact, run the task - that is already happening in Hydra at this point
-      download_task.run
+      unless etag.nil?
+        headers << ['If-None-Match', etag]
+      end
+
+      begin
+        netrc_info = Netrc.read
+        netrc_host = URI.parse(file_remote_url).host
+        credentials = netrc_info[netrc_host]
+        if credentials
+          user, pass = credentials
+          headers << ['Authorization', "Basic #{Base64.strict_encode64("#{user}:#{pass}")}"]
```

Comment:
You could also use https://github.com/socketry/protocol-http/blob/master/lib/protocol/http/header/authorization.rb

Reply:
Switch to `Protocol::HTTP::Header::Authorization.basic(user, pass)` thanks!